__tests__/workers/postUpdated.ts

```diff
@@ -557,6 +558,58 @@ it('save a post as private if source is private', async () => {
   expect(posts[2].flags.private).toBe(true);
 });
 
+it('do not save post if source can not be found', async () => {
+  await expectSuccessfulBackground(worker, {
+    id: 'f99a445f-e2fb-48e8-959c-e02a17f5e816',
+    title: 'Title',
+    url: 'https://post.com',
+    source_id: 'source-does-not-exist-on-the-api',
+  });
+  const post = await con.getRepository(Post).findOne({
+    where: {
+      id: 'f99a445f-e2fb-48e8-959c-e02a17f5e816',
+    },
+  });
+  expect(post).toBeNull();
+});
+
+it('should save post if source_id is not passed but post exists', async () => {
+  const newPostId = await generateShortId();
```

Comment:
test to support https://github.com/dailydotdev/daily-api/pull/2609/files#r1926068941